connect-inject/annotations.go

```diff
@@ -84,9 +84,9 @@ const (
 	// annotationConsulNamespace is the Consul namespace the service is registered into.
 	annotationConsulNamespace = "consul.hashicorp.com/consul-namespace"
 
-	// annotationTransparentProxy enables or disables transparent proxy mode for a given pod.
+	// annotationTransparentProxy enables or disables default transparent proxy behaviour for a given pod or namespace.
 	// This annotation takes a boolean value (true/false).
-	annotationTransparentProxy = "consul.hashicorp.com/transparent-proxy"
+	annotationTransparentProxy = "consul.hashicorp.com/transparent-proxy-default"
```

Comment:
Curious about this change. In my mind it makes sense as `transparent-proxy: true`. I'm guessing you're thinking that if I have it on a namespace *and* on the pod then the namespace annotation is only the default. But when it's on a pod it's not the default, it's the actual value that gets used.

We *could* have `transparent-proxy-default: true` for the namespace and `transparent-proxy: true` for the pod but I think this is probably too complicated. I think most users will understand `transparent-proxy: true` at the namespace level can be overridden at the pod level.

Reply:
I'm with you on this point. It was suggested as a means to make it a bit more descriptive of the behaviour at the namespace level, but I think it's creating a bit more confusion than clarity. Totally fine with moving this back to how it was!

Reply:
Yep, @lkysow you're absolutely right. This should only apply to the namespace label since you could still override the namespace setting with a pod annotation. It doesn't make sense to have `-default` for the pod annotation since there isn't anything else that could override it.

Reply:
>We could have transparent-proxy-default: true for the namespace and transparent-proxy: true for the pod but I think this is probably too complicated. I think most users will understand transparent-proxy: true at the namespace level can be overridden at the pod level.

The reason we thought it makes sense for the namespace is to make it similar the helm value (`transparentProxy.defaultEnabled`) in that the behavior is very similar between the two: one applies to the cluster defautl and the other to the namespace.

I think you're right that it might be intuitive for the users to just have `transparent-proxy` label but we wanted to make it explicit and consistent with the helm value.

Reply:
> I think you're right that it might be intuitive for the users to just have transparent-proxy label but we wanted to make it explicit and consistent with the helm value.

I think that it's more likely that users mess this up and put the wrong label on the pod or namespace. In Helm there's only one value `defaultEnabled`, not two: `enabled`, `defaultEnabled`. If we had both then I think users would mess it up there too.

I'd vote to stick with `transparent-proxy` for both namespace and pod.